### PR TITLE
Create `Client` Entity

### LINKED ISSUE #5: Create `Client` entity
### Description
Create `Client` entity

Infrastructure link - https://link.excalidraw.com/l/7x0yF4kXlsV/8ho7kx5CwvD

### Requirements
- project should be runnable

### Deliverables
- `Client` entity, 
- `Client` repository
- `Client` foundation service
- Fluent Valiation
- Add **temporary** seed data
- Add Migration
- Register all necessary infrastructure to DI

### PR DESCRIPTION
**Pull request by bunyodkamalov**
Closes #5

### CODE REVIEW

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Client.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+public class Client : AuditableEntity
+{
+    public string FirstName { get; set; } = default!;
+
+    public string LastName { get; set; } = default!;
+
+    public string Email { get; set; } = default!;
+
+    public string Password { get; set; } = default!;
+
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/Client.cs:14]: Add comments

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/DataContexts/AppDbContext.cs [hunk 2]
```diff
@@ -1,11 +1,14 @@
 ﻿using System.Reflection;
+using Feedback.Analyzer.Domain.Entities;
 using Microsoft.EntityFrameworkCore;
 
 namespace Feedback.Analyzer.Persistence.DataContexts;
 
 public class AppDbContext(DbContextOptions<AppDbContext> dbContextOptions) : DbContext(dbContextOptions)
 {
     //Db sets
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/DataContexts/AppDbContext.cs:9]: Remove this comment and add region with name "Client infrastructure"

**Review by rashidovasadbek — DISMISSED** [review]

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Configurations/ClientConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+namespace Feedback.Analyzer.Persistence.Configurations;
+
+public class ClientConfiguration : IEntityTypeConfiguration<Client>
+{
+    public void Configure(EntityTypeBuilder<Client> builder)
+    {
+        builder.Property(client => client.FirstName).HasMaxLength(256).IsRequired();
+        builder.Property(client => client.LastName).HasMaxLength(256).IsRequired();
+        builder.Property(client => client.Email).HasMaxLength(128).IsRequired();
+        builder.Property(client => client.Password).HasMaxLength(128).IsRequired();
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Configurations/ClientConfiguration.cs:14]: Change constraints as 

firstname - max 64
lastname - max 64

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IClientRepository.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    ValueTask<Client?> GetByIdAsync(Guid clientId, bool asNoTracking = false, CancellationToken cancellationToken = default);
+
+    ValueTask<Client> CreateAsync(Client client, bool saveChanges = true, CancellationToken cancellationToken = default);
+
+    ValueTask<Client> UpdateAsync(Client client, bool saveChanges = true, CancellationToken cancellationToken = default);
+
+    ValueTask<Client?> DeleteAsync(Client client, bool saveChanges, CancellationToken cancellationToken = default);
+    
+    ValueTask<Client?> DeleteByIdAsync(Guid clientId, bool saveChanges = true, CancellationToken cancellationToken = default);
+}
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/Interfaces/IClientRepository.cs:19]: Add documentation comments

File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/ClientRepository.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq.Expressions;
+using System.Runtime.CompilerServices;
+using Feedback.Analyzer.Domain.Entities;
+using Feedback.Analyzer.Persistence.DataContexts;
+using Feedback.Analyzer.Persistence.Repositories.Interfaces;
+
+namespace Feedback.Analyzer.Persistence.Repositories;
+
+public class ClientRepository(AppDbContext dbContext) : EntityRepositoryBase<Client, AppDbContext>(dbContext), IClientRepository
```
Comment (sulton-max) [inline src/Feedback.Analyzer.Backend/Feedback.Analyzer.Persistence/Repositories/ClientRepository.cs:9]: Add only type docs

**Review by sulton-max — CHANGES_REQUESTED** [review]
Good job

**Review by sulton-max — DISMISSED** [review]
Good job

**Review by botirjonabdumannonovv — DISMISSED** [review]

**Review by sulton-max — APPROVED** [review]

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by Ilxom-22 — APPROVED** [review]

**Pull request merged**